Source/WebCore/dom/Document.cpp

```diff
@@ -833,18 +833,20 @@ Document::~Document()
 
     // End the loading signpost here in case loadEventEnd never fired.
     WTFEndSignpost(this, NavigationAndPaintTiming);
+
+    RELEASE_ASSERT(!m_referencingNodeCount);
```

Comment:
Aren't we supposed to fix the lambda capture in the animation code too? Otherwise, the high priority radar you're associating this PR with will still not be fixed if I understand correctly?

Reply:
The lambda capture is valid because Document supports resurrection from removedLastRef().

We only need to change the lambda capture if we want to reduce the occurrence of resurrection (which I think we should, in a follow-up, since resurrection will almost always indicate wasted work):

>         // FIXME: We could be stricter, and forbid escaping RefPtrs in the code
>          // above just like we forbid escaping RefPtrs in destructors. We still
>          // need to support resurrection for m_referencingNodeCount, but we can
>          // establish a simpler model that's less likely to cause leaks or
>          // wasted work.